### ISSUE TITLE
Won't update on new data

### ISSUE THREAD

**Issue by alexmcmillan** [nsmith7989/redux-filter#13]
When initially rendering with 10 `subjects` all works fine, but if you add more `subjects` they do not appear in the `collection` inside the wrapped component.

**Comment by nsmith7989** [comment]
Good point. I never thought about filtering dynamic subjects, but I can definitely see this being useful. 

There are a couple considerations we'd need to take into account. 
1. If the new subjects get change the possible options should those be updated as well? I think that makes sense. 
   1. At the moment building the options based on subjects happens in the [constructor of `Filter`](https://github.com/nsmith7989/redux-filter/blob/master/src/index.js#L26), so that logic will need to be moved into `componentWillReceiveProps` and then options rebuilt. 
   2. This is also where the redux store is being created. ATM, some of the store's initial state is based on what subjects are passed in. Obviously we can't just make a new store because we'd lose state. So there will need to be an action that adds to stores state when new subjects are added. 

TL;DR definitely possible, but will take some consideration.
